test/unit/org/apache/cassandra/tools/SSTableRepairedAtSetterTest.java

```diff
@@ -43,10 +56,73 @@ public void testSSTableRepairedAtSetter_NoArgs()
+        {
+            String help = "This command should be run with Cassandra stopped, otherwise you will get very strange behavior\n" + 
+                          "Verify that Cassandra is not running and then execute the command like this:\n" + 
+                          "Usage: sstablerepairedset --really-set [--is-repaired | --is-unrepaired] [-f <sstable-list> | <sstables>]\n";
+            Assertions.assertThat(tool.getStdout()).isEqualTo(help);
+        }
+    }
+
+    @Test
+    public void testWrongArgFailsAndPrintsHelp() throws IOException
+    {
+        try (ToolRunner tool = runner.invokeClassAsTool(SSTableRepairedAtSetter.class.getName(),
+                                                       "--debugwrong",
+                                                       findOneSSTable("legacy_sstables", "legacy_ma_simple")))
+        {
+            assertThat(tool.getStdout(), CoreMatchers.containsStringIgnoringCase("usage:"));
+            assertTrue(tool.getCleanedStderr(), tool.getCleanedStderr().isEmpty());
+            assertEquals(1,tool.getExitCode());
+        }
+    }
+
+    @Test
+    public void testIsrepairedArg() throws Exception
     {
-        runner.invokeClassAsTool("org.apache.cassandra.tools.SSTableRepairedAtSetter", "--really-set", "--is-repaired", findOneSSTable("legacy_sstables", "legacy_ma_simple"))
-              .waitAndAssertOnCleanExit();
+        try (ToolRunner tool = runner.invokeClassAsTool(SSTableRepairedAtSetter.class.getName(),
+                                                       "--really-set",
+                                                       "--is-repaired",
+                                                       findOneSSTable("legacy_sstables", "legacy_ma_simple")))
+        {
+            tool.waitAndAssertOnCleanExit();
+        }
+        assertNoUnexpectedThreadsStarted(null, OPTIONAL_THREADS_WITH_SCHEMA);
+        assertSchemaNotLoaded();
+        assertCLSMNotLoaded();
+        assertSystemKSNotLoaded();
+        assertKeyspaceNotLoaded();
+        assertServerNotLoaded();
+    }
+
+    @Test
+    public void testIsunrepairedArg() throws Exception
+    {
+        try (ToolRunner tool = runner.invokeClassAsTool(SSTableRepairedAtSetter.class.getName(), "--really-set", "--is-unrepaired", findOneSSTable("legacy_sstables", "legacy_ma_simple")))
+        {
+              tool.waitAndAssertOnCleanExit();
+        }
+        assertNoUnexpectedThreadsStarted(null, OPTIONAL_THREADS_WITH_SCHEMA);
+        assertSchemaNotLoaded();
+        assertCLSMNotLoaded();
+        assertSystemKSNotLoaded();
+        assertKeyspaceNotLoaded();
+        assertServerNotLoaded();
+    }
+
+    @Test
+    public void testFilesArg() throws Exception
+    {
+        String file = Files.write(Paths.get("./sstablelist.txt"), findOneSSTable("legacy_sstables", "legacy_ma_simple").getBytes()).getFileName().toAbsolutePath().toString();
```

Comment:
shouldn't write to the current path, nothing will clean this up; should stick to build directory or tmp.